pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -90,29 +82,60 @@ func (m *Monitor) osdStatus() error {
 		id := int(id64)
 
 		logger.Debugf("validating status of osd.%d", id)
-		_, tracked := m.lastStatus[id]
 
-		// No action on in/out cluster state is taken at this time.
-		status, _, err := osdDump.StatusByID(int64(id))
+		status, in, err := osdDump.StatusByID(int64(id))
 		if err != nil {
 			return err
 		}
 
 		if status != upStatus {
 			logger.Infof("osd.%d is marked 'DOWN'", id)
-			if tracked {
-				evalDownStatus(id)
-			} else {
-				m.lastStatus[id] = time.Now()
-			}
 		} else {
 			logger.Debugf("osd.%d is healthy.", id)
-			if tracked {
-				logger.Infof("osd.%d recovered, stopping tracking.", id)
-				delete(m.lastStatus, id)
-			}
+
+		}
+
+		if in != inStatus {
+			logger.Infof("osd.%d is marked 'OUT'", id)
+			m.outOSDs[strconv.Itoa(id)] = true
```

Comment:
Do we need to save the `out` OSDs into the map? We can just immediately call `handleOSDMarkedOut()` and remove the K8s resources, right? I don't think we need to batch these at the same time for all `out` OSDs.

Reply:
`ceph osd safe-to-destroy` command can take multiple osd ids and return the osd ids that are `safe-to-destroy`. So what I was thinking was to utilize this. After every 5 min make a map of all the out osds and pass them to `safe-to-destroy` command (i.e., to `OsdSafeToDestroy()` function). Also when the output of `OsdSafeToDestroy()` is deserialized into json it gives a struct having a list of int(list of osd ids that are safe to destroy).

Reply:
I saved the `out` OSDs into map based on https://github.com/rook/rook/pull/3117#discussion_r284426435

Reply:
Yeah, now that I see how simple we can make the code path, I'm ready to change the recommendation. It should be a rare operation to remove an OSD after it is safe to remove, so I don't expect we will gain anything by batching the operation. Simplifying the code is always my preference if we can. :)

Reply:
cool :+1: